### PR TITLE
Fixes #26720 - CV Publish: copy correct errata

### PR DESCRIPTION
**Pull request by parthaa**
This commit makes the following changes to the cv publish process
Before This Commit:
Errata would get copied over to the destination repo if ANY of the
packages belonging to the errata were copied.

After this Commit:
Errata would get copied over to the destination repo only if ALL
of the packages belonging to the errata were copied.

This commit also includes changes to hide modular errata in a content
view filter (since all modules and module rpms are copied over, all the
modular errata will also get copied automatically.)

### CODE REVIEW

File: app/models/katello/erratum.rb [hunk 1]
```diff
@@ -110,6 +120,10 @@ def self.list_filenames_by_clauses(repo, clauses)
           where(statement).pluck(:filename)
     end
 
+    def modular?
+      ::Katello::Erratum.modular.where(:id => id).exists?
+    end
```
Comment (jlsherrill) [inline app/models/katello/erratum.rb]: why is this method needed?  what exactly is the intent?

Comment (parthaa) [reply app/models/katello/erratum.rb]: Good call. Removing it. I was planning to show if an errata has modules or not. But that can be done in another way if needed so removing it.

File: app/models/katello/repository.rb [hunk 2]
```diff
@@ -265,30 +265,40 @@ def product_type
...
+        errata_with_package_counts.keep_if { |id| errata_with_package_counts[id] == errata_with_packages_in_library[id] }
+        unless errata_with_package_counts.empty?
+          partial_errata = self.errata.where("#{Katello::Erratum.table_name}.id NOT IN (?)", errata_with_package_counts.keys)
+        end
       end
+
+      partial_errata
     end
 
     def empty_errata!
```
Comment (jlsherrill) [inline app/models/katello/repository.rb]: should this method get renamed?

**Comment by jlsherrill** [comment]
few things, but overall looks good. now to do some testing

**Comment by jlsherrill** [comment]
Tested a few scenarios and I'm happy with the results. Just a few small comments

**Comment by parthaa** [comment]
[test katello]

**Pull request merged**